completions/gcc

```diff
@@ -53,7 +53,7 @@ _comp_cmd_gcc()
         prefix_length=$((${#prefix} + 1))
         local flags=$("$1" --completion="$argument" | cut -c $prefix_length-)
         [[ ${flags} == "=*" ]] && compopt -o nospace 2>/dev/null
-        COMPREPLY=($(compgen -W "$flags" -- ""))
+        _comp_compgen -R -- -W "$flags"
```

Comment:
This PR changes some (but not all) cases of `-W` arg to use single quotes. Per https://github.com/scop/bash-completion/pull/961#discussion_r1186844166, should we do it everywhere?
```suggestion
        _comp_compgen -R -- -W '$flags'
```

Reply:
It is related also to https://github.com/scop/bash-completion/issues/958#issuecomment-1537214890. I sometimes thought we can keep the *existing* cases to address it all at once later but sometimes wanted to apply the change with another chance to the line. It is admittedly not so consistent. Probably I would like to revert it to the double quotes for now to later address it in a batch PR of similar cases.

Reply:
On a second thought, let me resolve those quote comments so I can see if something's ready to be merged already.

Reply:
I reverted it https://github.com/scop/bash-completion/compare/95eebfc73e82b15693079da1f41d2420eeac1aed..4358f1e5a1e3f9a3a292b6d2b7a7f27f1611907e